lib/facility_creator.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# # Parses API data and creates facilities
+class FacilityCreator
+  def create_facilities(facilities)
+    facilities.map do |facility|
+      Facility.new({
+                     name: parse_name(facility),
+                     address: parse_address(facility),
+                     phone: parse_phone(facility),
+                     hours: parse_hours(facility),
+                     holidays_closed: facility[:holidaysclosed]
+                   })
+    end
+  end
+
+  def parse_name(facility)
+    if facility.include?(:office_name)
+      facility.fetch_values(:office_name, :office_type).map do |name_array|
+        name_array.split.map(&:capitalize)
+      end.join(' ')
+    elsif facility.include?(:dmv_office)
+      facility[:dmv_office]
+    elsif facility.include?(:name)
+      facility[:name]
+    end
+  end
+
+  def parse_address(facility)
+    "#{unless facility[:street_address_line_1].nil?
```

Comment:
This block of code is not very readable. I would much rather a more multi-line process that extracts each component as a variable and then maybe joins them at the end or uses string interpolation to combine the data together. Anytime you're using `\` to join long strings together, it generally could indicate unreadable string manipulation